### PR TITLE
DOC: update the pandas.Index.is_categorical docstring

### PR DESCRIPTION
**Pull request by mariadelmarbibiloni**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [ ] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
################################## Validation ##################################
################################################################################

Errors found:
        See Also section not found

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

We think that See Also is not necessary for this function.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
What's the output of of the vlaidate_docstrings script?

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -1471,6 +1471,40 @@ def is_object(self):
         return is_object_dtype(self.dtype)
 
     def is_categorical(self):
+        """
+        Check if the index type is categorical.
+
+        The given object must be a pandas dataframe or pandas Series.
```
Comment (TomAugspurger) [inline pandas/core/indexes/base.py]: dataframe -> DataFrame

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -1471,6 +1471,40 @@ def is_object(self):
...
+
+        Returns
+        ----------
+        boolean
+            True if index is categorical.
+
+        Examples
+        --------
+        >>> s = pd.Series(["Watermelon","Orange","Apple",
```
Comment (jbigatti) [inline pandas/core/indexes/base.py]: Space after ","

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -1471,6 +1471,40 @@ def is_object(self):
         return is_object_dtype(self.dtype)
 
     def is_categorical(self):
+        """
+        Check if the index type is categorical.
+
+        The given object must be a pandas dataframe or pandas Series.
+
+        Returns
+        ----------
```
Comment (TomAugspurger) [inline pandas/core/indexes/base.py]: This should be the same length as "Returns"

File: pandas/core/indexes/base.py (hunk 2 above)
Comment (TomAugspurger) [inline pandas/core/indexes/base.py]: space after the commas

File: pandas/core/indexes/base.py [hunk 4]
```diff
@@ -1471,6 +1471,40 @@ def is_object(self):
...
+        boolean
+            True if index is categorical.
+
+        Examples
+        --------
+        >>> s = pd.Series(["Watermelon","Orange","Apple",
+        ...                 "Watermelon"]).astype('category')
+        >>> df = pd.DataFrame({'Weight (grams)':[2000,230,160,1300]}, index=s)
```
Comment (TomAugspurger) [inline pandas/core/indexes/base.py]: spcae after the colon. Space after commas.

File: pandas/core/indexes/base.py [hunk 5]
```diff
@@ -1471,6 +1471,40 @@ def is_object(self):
...
+        >>> df
+                    Weight (grams)
+        Watermelon            2000
+        Orange                 230
+        Apple                  160
+        Watermelon            1300
+        >>> df.index.is_categorical()
+        True
+
+        >>> df = pd.Series(["Peter","Adam","Elisabeth","Margareth"])
```
Comment (TomAugspurger) [inline pandas/core/indexes/base.py]: call the `ser = ` since it's a series :)

**Review by TomAugspurger — COMMENTED** [review]
Minor things if you could update. looks good otherwise!

File: pandas/core/indexes/base.py (hunk 4 above)
Comment (jbigatti) [inline pandas/core/indexes/base.py]: Space after ","

**Review by jbigatti — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 4 above)
Comment (mariadelmarbibiloni) [reply pandas/core/indexes/base.py]: Thank you for your comments ^^ !! We have already changed it.

Comment (mariadelmarbibiloni) [reply pandas/core/indexes/base.py]: Thanks, we appreciate your comments!  We have already changed it :)

File: pandas/core/indexes/base.py [hunk 6]
```diff
@@ -1471,6 +1471,41 @@ def is_object(self):
         return is_object_dtype(self.dtype)
 
     def is_categorical(self):
+        """
+        Check if the Index type is categorical.
+
+        The given object must be a pandas DataFrame or pandas Series.
```
Comment (jreback) [inline pandas/core/indexes/base.py]: this only applies to a Series or an Index, IOW it must be 1D

File: pandas/core/indexes/base.py [hunk 7]
```diff
@@ -1471,6 +1471,41 @@ def is_object(self):
...
+        Returns
+        -------
+        boolean
+            True if index is categorical.
+
+        Examples
+        --------
+        >>> s = pd.Series(["Watermelon", "Orange", "Apple",
+        ...                 "Watermelon"]).astype('category')
+        >>> df = pd.DataFrame({'Weight (grams)': [2000, 230, 160, 1300]},
```
Comment (jreback) [inline pandas/core/indexes/base.py]: don't create a DataFrame here just to use its index, just use the series above

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/base.py (hunk 7 above)
Comment (mariadelmarbibiloni) [reply pandas/core/indexes/base.py]: Ok, Thank you! I have changed the examples.

File: pandas/core/indexes/base.py (hunk 6 above)
Comment (mariadelmarbibiloni) [reply pandas/core/indexes/base.py]: Thanks @jreback ! I have changed the exended summary. Is it better now?

File: pandas/core/indexes/base.py [hunk 8]
```diff
@@ -1471,6 +1471,38 @@ def is_object(self):
         return is_object_dtype(self.dtype)
 
     def is_categorical(self):
+        """
+        Check if the Index type is categorical.
+
+        The given object must be a pandas DataFrame index or a pandas Series
```
Comment (TomAugspurger) [inline pandas/core/indexes/base.py]: I think it's best to just remove this. There isn't really an object being passed, it's just an attribute on Index classes.

**Comment by TomAugspurger** [comment]
Just pushed a commit removing the extended sumary. Added a See Also for CategoricalIndex.

File: pandas/core/indexes/base.py [hunk 9]
```diff
@@ -1472,15 +1472,16 @@ def is_object(self):
...
 
         Returns
         -------
         boolean
-            True if index is categorical.
+            True if the Index is categorical.
+
+        See Also
+        --------
+        CategoricalIndex : Index for categorical data.
```
Comment (mariadelmarbibiloni) [inline pandas/core/indexes/base.py]: Thanks a lot for the See Also!

File: pandas/core/indexes/base.py [hunk 10]
```diff
@@ -1472,15 +1472,16 @@ def is_object(self):
 
     def is_categorical(self):
         """
-        Check if the Index type is categorical.
-
-        The given object must be a pandas DataFrame index or a pandas Series
-        index.
+        Check if the Index holds categorical data.
```
Comment (mariadelmarbibiloni) [inline pandas/core/indexes/base.py]: I agree! Thank you :)

**Comment by TomAugspurger** [comment]
Thank you for working on this.

**Pull request merged**